### PR TITLE
[ADAM-1793] Adding vcf2adam and adam2vcf that handle separate variant and genotype data.

### LINKED ISSUE #1793: Add `transformVariantContexts` or similar to cli
Per #1771 it would be useful to load VCF and save both `Variant`s and `Genotype`s separately to Parquet+Avro.

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #1793 

Rough first cut, needs review.

### CODE REVIEW

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2Adam.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  def run(sc: SparkContext) {
+    require(!(args.sort && args.sortLexicographically),
+      "Cannot set both -sort_on_save and -sort_lexicographically_on_save.")
+
+    val variantContexts = sc.loadVcf(
+      args.inputPath,
+      stringency = stringency)
+
+    // todo: cache variantContexts?  sort variantContexts first?
```
Comment (fnothaft) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2Adam.scala]: I would present this as an option. It's fine to convert twice if you're working with a small dataset, but you'll want to persist if you're working with a large data.

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Adam2Vcf.scala [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      stringency = stringency)
+
+    val genotypes = sc.loadGenotypes(
+      args.genotypesInputPath,
+      optPredicate = None,
+      optProjection = None,
+      stringency = stringency)
+
+    val join = variants.shuffleRegionJoin(genotypes)
+    val updatedGenotypes = join.rdd.map(pair => {
```
Comment (fnothaft) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Adam2Vcf.scala]: Reading through this a second time, what I'd do here is not a shuffleRegionJoin (think of the multiallelic sites) but instead a join on contig/start/end/ref/alt. You might be able to do this through the dataset API for better (?) performance.

**Comment by heuermh** [comment]
Move to close this in favor of #1865.

**Comment by heuermh** [comment]
Closing in favor of #1865.

**Pull request closed without merging**